eZ/Publish/Core/REST/Server/Output/ValueObjectVisitor/RestExecutedView.php

```diff
@@ -115,7 +111,7 @@ public function visit(Visitor $visitor, Generator $generator, $data)
                 $contentInfo = $searchHit->valueObject->contentInfo;
                 $valueObject = new RestContentValue(
                     $contentInfo,
-                    $this->locationService->loadLocation($contentInfo->mainLocationId),
+                    $this->locationResolver->resolveLocation($contentInfo),
```

Comment:
In retrospect this is a bit problematic, the value is signified as main location:
```
     * @param \eZ\Publish\API\Repository\Values\Content\Location|null $mainLocation
```

If user does not have access we are supposed to pass null instead.

Reply:
@andrerom shouldn't we change the name of the property of `RestContent`in this case? So `mainLocation` -> `location` or would it be BC break?

Reply:
>shouldn't we change the name of the property of RestContentin this case? 

I would say we should rather add new property then for those cases, and keep logic with `null`.

Reply:
Maybe indeed like above. But to better know what to do, maybe check in what context it is needed to better define the domain context you need to fill.